tendrl/alerting/handlers/__init__.py

```diff
@@ -0,0 +1,224 @@
+import etcd
+import importlib
+import inspect
+import json
+import multiprocessing
+import os
+import six
+from tendrl.alerting import constants
+from tendrl.alerting.objects.alert import AlertUtils
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.utils.time_utils import now
+import time
+
+
+class NoHandlerException(Exception):
+    pass
+
+
+class HandlerMount(type):
+
+    def __init__(cls, name, bases, attrs):
+        if not hasattr(cls, 'handlers'):
+            cls.handlers = []
+        else:
+            cls.register_handler(cls)
+
+    def register_handler(cls, handler):
+        instance = handler()
+        cls.handlers.append(instance)
+
+
+@six.add_metaclass(HandlerMount)
+class AlertHandler(object):
+
+    handles = ''
+    representive_name = ''
+
+    def __init__(self):
+        self.time_stamp = now()
+        self.alert = None
+
+    def update_alert(self):
+        # Fetch alerts in etcd
+        try:
+            alerts = tendrl_ns.central_store_thread.get_alerts()
+            # Check if similar alert already exists
+            for curr_alert in alerts:
+                # If similar alert exists, update the similar alert to etcd
+                if AlertUtils().is_same(self.alert, curr_alert):
+                    self.alert = AlertUtils().update(
+                        self.alert,
+                        curr_alert
+                    )
+                    if not AlertUtils().equals(
+                        self.alert,
+                        curr_alert
+                    ):
+                        self.alert.save()
+                    return
+                # else add this new alert to etcd
+            self.alert.save()
+        except etcd.EtcdKeyNotFound:
+            try:
+                self.alert.save()
+            except Exception as ex:
+                Event(
```

Comment:
I feel @GowthamShanmugam is trying to have a class defined using which exception can be logged. Check once